### PR TITLE
EZP-29170: Custom tag attribute of "link" type should indicate what node is selected #1362

### PR DESCRIPTION
**Pull request by thiagocamposviana**
https://jira.ez.no/browse/EZP-29170

This is pull request replaces https://github.com/ezsystems/ezpublish-legacy/pull/1362

Now custom tags properly supports multiple link attributes on multiple tags.

Fixed issues since last pull request:

- Identified one issue when changing the link source type to eznode the code would throw errors.
- Identified one issue when editing objects with multiple customtags with multiple link types the code would throw errors when editing a custom tag containing multiple links.
- Identified one issue where link source would be pre-populated with content from other link types that had already an existing source.

Original description. 

The first time you select a node in a custom tag "link" attribute, it shows you the name of the node selected. However, if you open the dialog again, it does not show you the name. To an editor, "eznode://66214" is not user-friendly at all. It should at the very least show the same information it did when the node was first selected.

Testing Notes (just a sample custom tag you can define so you can test this):

```
settings/override/content.ini.append.php

[CustomTagSettings]
....
AvailableCustomTags[]=test_link

...

[test_link]
CustomAttributes[]=name
CustomAttributes[]=link1
CustomAttributes[]=link2


settings/override/ezoe_attributes.ini.append.php


[CustomAttribute_test_link_name]
Name=Name
Title=Name
Type=text
Required=false
 

[CustomAttribute_test_link_link1]
Name=Link test
Title=Link test
Type=link
Required=false
 
[CustomAttribute_test_link_link2]
Name=Link test2
Title=Link test2
Type=link
Required=false


```

Repeat those settings changing the custom tag name to get a proper set up for testing.

### CODE REVIEW

**Comment by thiagocamposviana** [comment]
There is still one issue I am investigating about the ezoe links (the href button).

**Comment by thiagocamposviana** [comment]
The pending link issue should be fixed.

File: extension/ezoe/design/standard/templates/ezoe/customattributes/link.tpl [hunk 1]
```diff
@@ -127,23 +130,31 @@ eZOEPopupUtils.settings.browseLinkGenerator = function( n, mode, ed )
 // override so selected element is redirected to link input
 eZOEPopupUtils.selectByEmbedId = function( object_id, node_id, name )
 {
-    var lid = ezoeLinkAttribute.id, drop = jQuery( '#'+lid+'_source_types'), inp = jQuery( '#'+lid+'_source' ), info = jQuery( '#'+lid+'_source_info' )
+    var lid = ezoeLinkAttribute.id, drop = jQuery( '#'+lid+'_source_types'), inp = jQuery( '#'+lid+'_source' ), info = jQuery( '#'+lid+'_source_info' );
```
Comment (glye) [inline extension/ezoe/design/standard/templates/ezoe/customattributes/link.tpl]: Well spotted. Ideally such unrelated fixes should be in a separate commit, but no big deal.

**Comment by thiagocamposviana** [comment]
thank you!

**Pull request merged**